NNPTPZ1/Program.cs

```diff
@@ -46,288 +46,127 @@ static void Main(string[] args)
 
-            /// <summary>
-            /// Constructor
-            /// </summary>
-            public Poly() => Coe = new List<Cplx>();
 
-            public void Add(Cplx coe) =>
-                Coe.Add(coe);
 
-            /// <summary>
-            /// Derives this polynomial and creates new one
-            /// </summary>
-            /// <returns>Derivated polynomial</returns>
-            public Poly Derive()
+        static int[] ParseIntegerArguments(string[] args)
+        {
+            int[] intargs = new int[2];
+            for (int i = 0; i < intargs.Length; i++)
             {
-                Poly p = new Poly();
-                for (int q = 1; q < Coe.Count; q++)
-                {
-                    p.Coe.Add(Coe[q].Multiply(new Cplx() { Re = q }));
-                }
-
-                return p;
+                intargs[i] = int.Parse(args[i]);
             }
+            return intargs;
+        }
 
-            /// <summary>
-            /// Evaluates polynomial at given point
-            /// </summary>
-            /// <param name="x">point of evaluation</param>
-            /// <returns>y</returns>
-            public Cplx Eval(double x)
+        static double[] ParseDoubleArguments(string[] args)
+        {
+            double[] doubleargs = new double[4];
+            for (int i = 0; i < doubleargs.Length; i++)
             {
-                var y = Eval(new Cplx() { Re = x, Imaginari = 0 });
-                return y;
+                doubleargs[i] = double.Parse(args[i + 2]);
             }
+            return doubleargs;
+        }
 
-            /// <summary>
-            /// Evaluates polynomial at given point
-            /// </summary>
-            /// <param name="x">point of evaluation</param>
-            /// <returns>y</returns>
-            public Cplx Eval(Cplx x)
-            {
-                Cplx s = Cplx.Zero;
-                for (int i = 0; i < Coe.Count; i++)
-                {
-                    Cplx coef = Coe[i];
-                    Cplx bx = x;
-                    int power = i;
-
-                    if (i > 0)
-                    {
-                        for (int j = 0; j < power - 1; j++)
-                            bx = bx.Multiply(x);
-
-                        coef = coef.Multiply(bx);
-                    }
-
-                    s = s.Add(coef);
-                }
-
-                return s;
-            }
+        static Polynomial CreatePolynomial()
+        {
+            Polynomial p = new Polynomial();
+            p.Coefficients.Add(new ComplexNumber() { Real = 1 });
+            p.Coefficients.Add(ComplexNumber.Zero);
+            p.Coefficients.Add(ComplexNumber.Zero);
+            p.Coefficients.Add(new ComplexNumber() { Real = 1 });
+            return p;
+        }
+        static void GenerateFractalImage(int width, int height, double xmin, double ymin, double xstep, double ystep, Polynomial p, Polynomial pd, Bitmap bmp)
```

Comment:
metoda uvnitř pojímá několik různých úrovní abstrakce, sjednoťte to prosím na jednu úroveň abstrakce, není vhodné kombinovat abstrakce a detaily